edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs

```diff
@@ -459,130 +461,126 @@ async Task PruneServiceAccounts(List<string> accountNamesToPrune)
+                    image: this.proxyImage,
+                    volumeMounts: proxyMounts)
+            };
 
-                var containerList = new List<V1Container>()
+            Option<List<V1LocalObjectReference>> imageSecret = moduleWithDockerConfig.Config.AuthConfig.Map(
+                auth =>
                 {
-                    new V1Container(
-                        KubeUtils.SanitizeDNSValue(moduleIdentity.ModuleId),
-                        env: env,
-                        image: moduleImage,
-                        volumeMounts: volumeMountList,
-                        securityContext: securityContext.GetOrElse(() => null),
-                        ports: exposedPortsOption.GetOrElse(() => null)),
-
-                    // TODO: Add Proxy container here - configmap for proxy configuration.
-                    new V1Container(
-                        "proxy",
-                        env: env, // TODO: check these for validity for proxy.
-                        image: this.proxyImage,
-                        volumeMounts: proxyMounts)
-                };
-
-                Option<List<V1LocalObjectReference>> imageSecret = moduleWithDockerConfig.Config.AuthConfig.Map(
-                    auth =>
+                    var secret = new ImagePullSecret(auth);
+                    var authList = new List<V1LocalObjectReference>
                     {
-                        var secret = new ImagePullSecret(auth);
-                        var authList = new List<V1LocalObjectReference>
-                        {
-                            new V1LocalObjectReference(secret.Name)
-                        };
-                        return authList;
-                    });
-
-                var modulePodSpec = new V1PodSpec(containerList, volumes: volumeList, imagePullSecrets: imageSecret.GetOrElse(() => null));
-                var objectMeta = new V1ObjectMeta(labels: podLabels, annotations: podAnnotations);
-                return new V1PodTemplateSpec(metadata: objectMeta, spec: modulePodSpec);
-            }
-            else
-            {
-                Events.InvalidModuleType(module);
-            }
+                        new V1LocalObjectReference(secret.Name)
+                    };
+                    return authList;
+                });
 
-            return new V1PodTemplateSpec();
+            var modulePodSpec = new V1PodSpec(containerList, volumes: volumeList, imagePullSecrets: imageSecret.OrDefault());
+            var objectMeta = new V1ObjectMeta(labels: podLabels, annotations: podAnnotations);
+            return new V1PodTemplateSpec(objectMeta, modulePodSpec);
         }
 
-        private VolumeOptions GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> moduleWithDockerConfig)
+        (List<V1Volume>, List<V1VolumeMount>, List<V1VolumeMount>) GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> moduleWithDockerConfig)
```

Comment:
Rather than put it back as a three tuple, why not create a VolumeOptions type?

Reply:
We don't use option here at all. Private type for private method seems like overkill or not?